src/main/scala/goatrodeo/util/ArtifactWrapper.scala

```diff
@@ -27,59 +33,32 @@ sealed trait ArtifactWrapper {
     *
     * @return
     */
-  def asStream(): BufferedInputStream
+  def asStream(): InputStream
 
   /** The name of the Artifact. This corresponds to the name of a `File` on disk
     *
     * @return
     */
   def path(): String
 
-  /** The number of bytes in the artifact. This is
+  /** The number of bytes in the artifact.
     *
     * @return
     */
   def size(): Long
 
-  /** Corresponds to `File.isFile()`
-    *
-    * @return
-    */
-  def isFile(): Boolean
+  private lazy val _mimeType: String =
+    ArtifactWrapper.mimeTypeFor(this.asStream(), this.path())
 
-  /** Corresponds to `File.isDirectory()`
-    *
-    * @return
-    */
-  def isDirectory(): Boolean
+  def mimeType: String = _mimeType
 
-  /** Get all the `ArtifactWrappers` for the entities in a directory. This will
-    * only return something if the ArtifactWrapper is a wrapper around `File`
-    * and the entity being traversed is a real filesystem.
-    *
-    * For "container" file types (e.g., a TAR), see
-    * `FileWalker.streamForArchive`
-    *
-    * @return
-    */
-  def listFiles(): Iterator[ArtifactWrapper]
+  lazy val uuid = UUID.randomUUID().toString()
```

Comment:
Would there ever be a need to instantiate an `ArtifactWrapper` with a known `uuid`?

Reply:
At this point, I can't think of one.

In the future, if we need to define the UUID for ArtifactWrapper, I think we can enhance the classes